### ISSUE TITLE
Optionen für Waschmaschine funktionieren nicht (am Beispiel Siemens WG56B2090/01)

### ISSUE THREAD

**Issue by Andersmacher** [iobroker-community-adapters/ioBroker.homeconnect#128]
In der Adapter/Instanzbeschreibung steht unter Absatz "Benutzung":
"Ändern des States programs.selected.BSH_Common_Root_SelectedProgram führt zum auswählen des Programms **![Unbenannt](https://github.com/iobroker-community-adapters/ioBroker.homeconnect/assets/88731624/5b203d7e-bd78-4d18-a1d0-e0e8a4ce91cf)**"

Das Auswählen einens Programms (z. B. Baumwolle) funktioniert durch setzen von "homeconnect.0.haID des Geräts.programs.selected.BSH_Common_Root_SelectedProgram" auf "LaundryCare.Washer.Program.Cotton". Ebenso funktioniert dies für andere Programme.

Jetzt zum Problem:
1)
Für das Auswählen/Ändern einer Temperatur/Schleuderdrehzahl gibt es keinen Datenpunkt wie z. B. "homeconnect.0.haID des Geräts.programs.selected.BSH_Common_Root_SelectedOption" oder ähnlich. Es gibt jedoch unter "homeconnect.0.haID des Geräts.programs.selected.options" die Datenpunkte "LaundryCare_Washer_Option_Temperature" und "LaundryCare_Washer_Option_SpinSpeed". Ein Setzen dieser Datenpunkte (z. B. auf "LaundryCare.Washer.EnumType.Temperature.GC30" oder "LaundryCare.Washer.EnumType.SpinSpeed.RPM800" führt jedoch nicht zur tatsächlichen Auswahl dieser Werte in der Maschine.
Setzt man hingegen "homeconnect.0.haID des Geräts.programs.selected.Cotton.LaundryCare_Washer_Option_Temperature" oder "homeconnect.0.haID des Geräts.programs.selected.Cotton.LaundryCare_Washer_Option_SpinSpeed" auf diese Werte, wird dieser Wert auch an der Maschine tatsächlich ausgewählt, wobei man Temperatur/Schleuderdrehzahl für ein anderes Programm (z. B. DarkWash) ebenso durch setzen der Option unter ".../Cotton/..." einstellen kann. Somit ist es zwar grundsätzlich möglich, die Option für Temperatur/Schleuderdrehzahl einzustellen, jedoch erscheint die Methode nicht plausibel/gewollt, zumindest ist sie verwirrend und sollte in der Anleitung zur Einstellung des/der Adapters/Instanz beschrieben werden.
2)
Keine der anderen Optionen (wie z. B. "LaundryCare_Washer_Option_Prewash", "LaundryCare_Washer_Option_LessIroning", ...) läßt sich durch Setzen irgendeines Datenpunktes auf was auch immer setzen.

**Comment by Bernd9000** [comment]
Jo, ist bei mir genau so, irgendwas macht die Home connect App anders weil dort eingestellte Temperaturen oder Drehzahlen  sofort übernommen werden.Hat jemmand eine Lösung ?

**Comment by Bernd9000** [comment]
Ich hab das jetzt nochmal getestet, alle Datenpunkte vorher gelöscht und Adapter neu gestartet.
In folgenden Pfaden habe ich die Optionen für u.a. Temperatur und Schleuderdrehzahl.

`*.programs.active.options`          (erst wenn Programm gestartet wurde)
`*.programs.available.options`
`*.programs.selected.options`
`*.programs.selected.options.Cotton`     (und alle weiteren Waschprogramme)

Wird ein Waschprogramm direkt an der Waschmaschine eingestellt landen diese Einstellungen erstmal unter:
*.programs.selected.options
Startet man das Programm bei der Waschmaschine wird alles im folgenden Pfad abgelegt:
*.programs.active.options

Über den Adapter/API funktioniert das nicht.
Es werden dann nur die Standartwerte in `*.programs.active.options` eingetragen.

Der Trick die Werte in `Cotton` einzutragen funktioniert bei mir jetzt nicht mehr.
Das einzige was funktioniert ist das Programm zu starten über `*.programs.active.BSH_Common_Root_ActiveProgram`
und dann nachträglich die Werte in `*.programs.active.options` zu überschreiben.

Ich frage mich jetzt liegt das an dem Adapter oder an der API ?

**Comment by Andersmacher** [comment]
Nur vorsorglich, falls das bei meiner Issue-Beschreibung (siehe ganz oben) nicht so klar herauskam:

Auch wenn das Einstellen von Temperatur und / oder Schleuderdrehzahl via ioBroker vielleicht nicht „intuitiv“ ist, grundsätzlich funktioniert es bei mir. Das ist also nicht der Grund für das von mir angelegte Issue, sondern nur ein Hinweis auf etwaige Verbesserungsmöglichkeiten.

**Das eigentliche Issue ist:**
Alle anderen Optionen (also alle AUßER Temperatur und / oder Schleuderdrehzahl) lassen sich **gar nicht** via ioBroker einstellen.
